src/Microsoft.AspNetCore.Mvc.TagHelpers/Cache/MemoryHtmlFragmentCache.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Cache
+{
+    public class MemoryHtmlFragmentCache : IHtmlFragmentCache
+    {
+        private const string PriorityOptionName = "priority";
+
+        private readonly IMemoryCache _memoryCache;
+
+        public MemoryHtmlFragmentCache(IMemoryCache memoryCache)
+        {
+            _memoryCache = memoryCache;
+        }
+
+        public async Task<IHtmlContent> SetAsync(object key, Func<Task<IHtmlContent>> renderContent, HtmlFragmentCacheContext context)
+        {
+            var options = new MemoryCacheEntryOptions();
+
+            if (context.ExpiresOn != null)
+            {
+                options.SetAbsoluteExpiration(context.ExpiresOn.Value);
+            }
+
+            if (context.ExpiresAfter != null)
+            {
+                options.SetAbsoluteExpiration(context.ExpiresAfter.Value);
+            }
+
+            if (context.ExpiresSliding != null)
+            {
+                options.SetSlidingExpiration(context.ExpiresSliding.Value);
+            }
+
+            if (context.HasOptions && context.Options.ContainsKey(PriorityOptionName))
+            {
+                CacheItemPriority priority;
+                if (Enum.TryParse(context.Options[PriorityOptionName], out priority))
+                {
+                    options.SetPriority(priority);
+                }
+            }
+
+            // Create an entry link scope and flow it so that any tokens related to the cache entries
+            // created within this scope get copied to this scope.
+            using (var link = _memoryCache.CreateLinkingScope())
+            {
+                var content = await renderContent();
+
+                options.AddEntryLink(link);
+
+                var stringBuilder = new StringBuilder();
+                using (var writer = new StringWriter(stringBuilder))
+                {
+                    content.WriteTo(writer, context.HtmlEncoder);
+                }
+
+                var result = new StringBuilderHtmlContent(stringBuilder);
+
+                _memoryCache.Set(key, result, options);
```

Comment:
It would be simpler actually, no scope to use/support. And yes, I want to add a default implementation in the module for `IDistributedCache`. I created the PR to allow comments on this phase. 
@Eilon should we have instead of two different implementations by default, a single one that we can switch from options on each tag helper, like Damian wanted initially? We could register the memory cache like now, and throw a meaningful exception if the `IDistributedCache` is not registered when the option is triggered.